aws/ecs_task_definition_equivalency.go

```diff
@@ -34,11 +37,14 @@ func ecsContainerDefinitionsAreEquivalent(def1, def2 string) (bool, error) {
 	if err != nil {
 		return false, err
 	}
+
 	canonicalJson2, err := jsonutil.BuildJSON(obj2)
 	if err != nil {
 		return false, err
 	}
 
+	log.Printf("[DEBUG] Comparing canonical definitions,\nFirst: %s\nSecond: %s\n",
+		canonicalJson1, canonicalJson2)
```

Comment:
nit: not sure if this is debugging that you left in unintentionally, however if intended, maybe we should log instead the result of the compare; `logical match` or `difference` etc. Right here we're just logging both and not actually saying if Terraform seems them as equal, that's left up to the ~user~ debugger

Reply:
right, good point - I'm happy to add more context (i.e. whether it's a match).

The reason I left it there is because I had hard times comparing those JSONs in a failing test, where you can only get the original (non-canonical) JSONs which is also kind of intention, because this process should remain hidden as implementation detail.